frontend/ui/widget/textboxwidget.lua

```diff
@@ -499,6 +500,25 @@ function TextBoxWidget:_splitToLines()
 
         ::idx_continue:: -- (Label for goto when use_xtext=true)
     end
+
+    if self.honor_soft_hyphen then
+        -- Check if there is a soft hyphen at the end of a line, replace it with a visible hyphen.
+        -- Begin from the end.
+        local soft_hyphens_at_eol = false
+        for i = #self.vertical_string_list, 1, -1 do
+            local pos = self.vertical_string_list[i].end_offset
+            if self.text:sub(pos, pos + 1) == "\xC2\xAD" then
+                self.text = self.text:sub(1, pos - 1) .. "-" .. self.text:sub(pos + 2)
+                soft_hyphens_at_eol = true
+            end
+        end
+
+        if soft_hyphens_at_eol then
+            self.text:gsub("\xC2\xAD","") -- remove all other soft hyphens
+            self:setText(self.text)
+            self:_computeTextDimensions()
```

Comment:
So, you replaced above softhyphens at end of line with a minus-hyphen, and here you remove all others (zero width, so no impact) - and you reshape the modified text with `self:_computeTextDimensions()`.

First issue: you're callling `_computeTextDimensions()` and so `_splitToLines()` while you're not yet out of the first `_computeTextDimensions()` - and that first call will do things when you're out of the second call. I don't know what can/may happen, but it's not clean. (And expensive.)

But more serious: you're not computing the same text, and the line breaks may not be where you met them in the first iteration. The real minus with a non-0-width will get accounted, and it's possible you'll break before it, because its width can't go into the target_width. Either it will be pushed at the start of the next line, or xtext will give the break of the previous opportunity, and your minus-hyphen will be shown at the end of the first word on 2nd line.
ie `This is a line#this is next line` and `#` being a 0-width soft hyphen, you may have:
`This is a line#` fitting the target_width, but
`This is a line-` not fitting.
And you may end up after your 2nd _computeTextDimensions() with one of these:
```
This is a line
- this is next
line
```
or
```
This is a
line- this is
next line
```
(and this messing with all the next lines and possibly showing many real-hyphens in the middle of lines).

You may not have witnesssed this when you were testing because you always had room to fit a small minus-hyphen in the splitted lines which were often smaller enough than targeted_width.
But it feels like it's not the proper solution to handle this.

And I feel - can't say before implemeting it and see any new issues - it should better be handled in xtext's `_makeLine()` (and may be `_shapeLine()`).
And if that's the way to do it, we don't need to have this supported when `not use_xtext` - there's already some stuff we don't handle in this obsolete way (which is there only for debugging/comparing).

Reply:
Thanks, I have drilled up xtext (added two lines of code), so all changes here (except honor_soft_hyphen) will disappear.
Short tests in German (including auto soft hyphen insertion for existing infomessages with cre) look promising. 
I will test the whole thing thoroughly on we and commit it then.